hnn_core/network.py

```diff
@@ -832,6 +903,320 @@ def _get_src_type_and_pos(self, gid):
+                ', got {}'.format(type(connectivity).__name__))
+
+        for conn in connectivity:
+            src_type, src_gid = conn['src_type'], conn['src_gid']
+            target_type, target_gid = conn['target_type'], conn['target_gid']
+            loc, receptor = conn['loc'], conn['receptor']
+            nc_dict = conn['nc_dict']
+
+            valid_keys = ['src_type', 'src_gid', 'target_type', 'target_gid',
+                          'loc', 'receptor', 'nc_dict']
+            for key in conn.keys():
+                if key not in valid_keys:
+                    raise IndexError(
+                        'connectivity key must be one of {}, '
+                        "got '{}'".format(valid_keys, key))
+
+            # Ensure gids in range of Network.gid_ranges
+            if not isinstance(src_gid, int):
+                raise TypeError(
+                    "connectivity['src_gid'] must be of type int, "
+                    'got {}'.format(type(src_gid).__name__))
+
+            if not isinstance(target_gid, int):
+                raise TypeError(
+                    "connectivity['target_gid'] must be of type int, "
+                    'got {}'.format(type(target_gid).__name__))
+
+            assert np.sum([src_gid in gid_range for
+                           gid_range in self.gid_ranges.values()]) == 1
+            assert np.sum([target_gid in gid_range for
+                           gid_range in self.gid_ranges.values()]) == 1
+
+            # Ensure string inputs
+            string_args = ['src_type', 'target_type', 'loc', 'receptor']
+            string_items = [src_type, target_type, loc, receptor]
+            for arg, item in zip(string_args, string_items):
+                if not isinstance(item, str):
+                    raise TypeError(
+                        "connectivity['{}'] must be of type str"
+                        ', got {}'.format(arg, type(item).__name__))
+
+            # Validate nc_dict
+            if not isinstance(nc_dict, dict):
+                raise TypeError(
+                    "connectivity['nc_dict'] must be of type dict"
+                    ', got {}'.format(type(nc_dict).__name__))
+            nc_keys = ['A_delay', 'A_weight', 'lamtha', 'threshold']
+            for key, item in nc_dict.items():
+                if key not in nc_keys:
+                    raise IndexError(
+                        'nc_dict key must be one of {}, '
+                        "got '{}'".format(nc_keys, key))
+                if not isinstance(item, (int, float)):
+                    raise TypeError(
+                        "nc_dict['{}'] must be of type int or float, "
+                        "got {}".format(key, type(item).__name__))
+
+        self.connectivity_list.extend(connectivity)
+
+    def clear_connectivity(self):
+        """Remove all connections defined in Network.connectivity_list"""
+        self.connectivity_list = list()
```

Comment:
superfluous unless you make `connectivity_list` a private attribute. I'd just go simple and remove it for now

Reply:
Might be worth making a private attribute (...but it would be good for users to query existing connections). 

In any case `net.connectivity_list` differs substantially from user defined dictionaries provided to `add_connection()`, so I'd prefer to discourage direct edits of `net.connectivity_list`